src/pages/Sidebar.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import * as TiSocial from 'react-icons/ti';
+import { useDispatch } from 'react-redux';
+import logo from '../images/logo.png';
+import { logoutUser } from '../redux/actions/auth';
+
+function Sidebar() {
+  const [sidebar, setSidebar] = useState(false);
+  const showSidebar = () => setSidebar(!sidebar);
+  const showPage = () => {
+    if (sidebar) setSidebar(!sidebar);
+  };
+  const dispatch = useDispatch();
+  const logout = () => {
+    showPage();
+    dispatch(logoutUser());
+  };
+
+  return (
+    <div className="text-slate-600 h-screen sm:border-r-2 ">
+
+      <button type="button" className="sm:hidden m-3 absolute" onClick={showSidebar}>
+        {!sidebar ? (
+          <svg
+            className="w-6 h-6 text-gray-500"
+            fill="none"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+            strokeWidth="4"
+            viewBox="0 0 24 24"
+            stroke="currentColor"
+          >
+            <path d="M4 5h16M4 12h16M4" />
+          </svg>
+        ) : (
+          <svg
+            className="w-6 h-6 text-gray-500"
+            fill="none"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+            strokeWidth="4"
+            viewBox="0 0 24 24"
+            stroke="currentColor"
+          >
+            <line x1="18" y1="6" x2="6" y2="18" />
+            <line x1="6" y1="6" x2="18" y2="18" />
+          </svg>
+        )}
+      </button>
+
+      <nav
+        className={
+        !sidebar ? 'sm:flex flex-col h-full hidden'
+          : 'flex flex-col h-full bg-white pt-7 sm:pt-0'
+      }
+      >
+        <Link className="m-4 hidden sm:flex" to="/">
+          <img src={logo} alt="logo" width={100} />
+        </Link>
+        <ul className="sm:ml-2 pt-2 pb-3 mt-6 text-center">
+          <li className="flex flex-col">
+            {[
+              ['HOUSES', '/'],
+              ['RESERVATIONS', '/reservations'],
+              ['ADD HOUSES', '/add_house'],
+              ['DELETE HOUSES', '/delete_house'],
+            ].map(([title, url]) => (
+              <Link
+                to={url}
+                key={title}
+                onClick={showPage}
+                className="text-sm font-semibold px-3 py-2 hover:bg-lime-500 hover:text-white w-full"
+              >
+                {title}
+              </Link>
```

Comment:
You're also using `Link` instead of `NavLink`. Please take a look at the [documentation for NavLink](https://reactrouter.com/docs/en/v6/components/nav-link).

Reply:
Thank you for pointing this out and for the descriptive video 😄 👍 please check it now.